crates/wast/src/binary.rs

```diff
@@ -335,15 +335,13 @@ impl Encode for Limits {
 
 impl Encode for MemoryType {
     fn encode(&self, e: &mut Vec<u8>) {
-        if self.shared {
-            e.push(0x03);
-            self.limits.min.encode(e);
-            // Handle a textual error here by deferring the validation error
-            // until later. This isn't great though and we should probably just
-            // make `Encode` fallible if this comes up somewhere else
-            self.limits.max.unwrap_or(0).encode(e);
-        } else {
-            self.limits.encode(e);
+        let flag_max = self.limits.max.is_some() as u8;
+        let flag_shared = self.shared as u8;
+        let flags = flag_max | (flag_shared << 1);
+        e.push(flags);
+        self.limits.min.encode(e);
+        if let Some(max) = self.limits.max {
+            max.encode(e);
```

Comment:
Out of curiosity, what was the motivator for this change?

Reply:
There's a test that constructs a `(memory 1 shared)` and asserts that the error message is that the maximum length is required for a shared memory. Before this change what would emit a maximum memory of 0, which would still fail but as a 'maximum must be greater than minimum error'. I thought the previous error message was more intuitive, so I added this change. It's a small thing, not very important.

[1] https://searchfox.org/mozilla-central/rev/278046367dab878316f60f0bd7f740cf73f3c447/js/src/jit-test/tests/wasm/memory-sharing.js#43